apollo-entity/src/main/java/com/spotify/apollo/entity/EntityCodec.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * -\-\-
+ * Spotify Apollo Entity Middleware
+ * --
+ * Copyright (C) 2013 - 2016 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.apollo.entity;
+
+import java.io.IOException;
+
+/**
+ * Interface for serializing and de-serializing entity types.
+ */
+public interface EntityCodec {
+
+  String defaultContentType();
+
+  <E> byte[] write(E entity, Class<? extends E> clazz) throws IOException;
```

Comment:
I had a short thought about not doing unnecessary  wrapping of bytebuffers for performance reasons. But this is not really any better since the api forces the user to allocate a buffer.

I could write this api in terms of input/output streams to make it possible to be more efficient in the middleware later. But there's so many more places in apollo where byte-copying needs to improve.

What do you say, should we start here, or keep it as is and take the cost later (probably apollo 2.0.0)